ivadomed/loader/dataset_group.py

```diff
@@ -0,0 +1,307 @@
+            # self.val_filename_pairs, self.test_filename_pairs
+            return
+
+        if complex_keyword == DataloaderKW.TRAINING_VALIDATION:
+            # modifying the self.train_filename_pairs, self.val_filename_pairs
+            pass
+
+        elif complex_keyword == DataloaderKW.TRAINING_TEST:
+            # modifying the self.train_filename_pairs, self.test_filename_pairs
+            pass
+
+        elif complex_keyword == DataloaderKW.TRAINING_VALIDATION_TEST:
+            # modifying the self.train_filename_pairs, self.val_filename_pairs, self.test_filename_pairs
+            pass
+
+        else:
+            raise ValueError("Unknown Complex Data Loader Type")
+
+    def parse_simple_spec_dict_for_filename_pairs(self, a_simple_loader_json: dict) -> list:
+        """Load the sub dictionary within the json file and return the dictionary"""
+
+        filename_pairs = []
+
+        if a_simple_loader_json is None:
+            return filename_pairs
+
+        if a_simple_loader_json.get(DataloaderKW.MISSING_FILES_HANDLE) == FileMissingHandleKW.SKIP:
+            drop_missing = True
+        else:
+            drop_missing = False
+
+        if DataloaderKW.IMAGE_GROUND_TRUTH in a_simple_loader_json.keys():
+            filename_pairs = self.validate_simple_dataset(a_simple_loader_json, drop_missing)
+
+        return filename_pairs
+
+    def validate_simple_dataset(self, loader_json: dict, drop_missing: bool) -> list:
+        """
+        Parse a JSON dictionary of simple type: i.e. train, validate, test etc. and validate it.
+        Args:
+            a_simple_dataset:
+
+        Returns:
+
+        """
+        filename_pairs = []
+
+        if not loader_json.get(DataloaderKW.IMAGE_GROUND_TRUTH):
+            raise KeyError("Expected V2 loader configuration files but the Image/Groundtruth key was not found in the loader JSON.")
+
+        list_image_ground_truth_pairs: list = loader_json.get(DataloaderKW.IMAGE_GROUND_TRUTH)
+        list_image_ground_truth_pairs_filtered: list = []
+        # Go through each subject
+        for a_subject_image_ground_truth_pair in list_image_ground_truth_pairs:
+
+            skip_subject_flag = False
+
+            # 2 lists: Subject List + Ground Truth list
+            assert len(a_subject_image_ground_truth_pair) == 2, \
+                f"Either subject images list or groundtruth files list are missing!"
+
+            # Validate and trim Subject files list
+            list_subject_specific_images: list = a_subject_image_ground_truth_pair[0]
+            if len(list_subject_specific_images) > self.n_expected_input:
+                logger.warning(f"HIGHER number of image files found than expected, for subject specification "
+                               f"{a_subject_image_ground_truth_pair}. Expected {self.n_expected_input} "
+                               f"but found {len(list_subject_specific_images)} from {list_subject_specific_images}")
```

Comment:
```suggestion
                               f"but found {len(list_subject_specific_images)}. Using image files {list_subject_specific_images[:self.n_expected_input]}.")
```

Just a suggestion to warn the user of which files are indeed used explicitly.